### PR TITLE
Upgrade Mithril to v2.0.3

### PR DESCRIPTION
**Pull request by GaneshSPatil**
Upgrade mithril. Fixes: #6752

### CODE REVIEW

File: server/webapp/WEB-INF/rails/webpack/helpers/spark_routes.ts [hunk 1]
```diff
@@ -319,7 +319,7 @@ export class SparkRoutes {
   }
 
   static apiPluginInfoPath(query: PluginInfoQuery) {
-    const queryString = m.buildQueryString(query);
+    const queryString = m.buildQueryString(query as any);
```
Comment (ketan) [inline server/webapp/WEB-INF/rails/webpack/helpers/spark_routes.ts]:
```
import Mithril from "mithril";
...
query as Mithril.Params
```

is more appropriate.

File: server/webapp/WEB-INF/rails/webpack/models/auth_configs/spec/auth_configs_crud_spec.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ describe("AuthorizationConfigurationCRUD", () => {
     expect(request.method).toEqual("DELETE");
     expect(request.data()).toEqual(toJSON({} as AuthConfigJSON));
     expect(request.requestHeaders.Accept).toEqual("application/vnd.go.cd.v1+json");
-    expect(request.requestHeaders["Content-Type"]).toEqual("application/json; charset=utf-8");
+    expect(request.requestHeaders["Content-Type"]).toEqual(undefined!);
```
Comment (ketan) [inline server/webapp/WEB-INF/rails/webpack/models/auth_configs/spec/auth_configs_crud_spec.ts]: why this change?

Comment (GaneshSPatil) [reply server/webapp/WEB-INF/rails/webpack/models/auth_configs/spec/auth_configs_crud_spec.ts]: As per the [mithril's changelog](https://github.com/MithrilJS/mithril.js/releases) it is stated that the request header `https://github.com/MithrilJS/mithril.js/releases` is set for all XHR methods by default, **provided they have a body `that's != null`**.

In case of delete, we don't send any request body, hence we are not required to set this header.
This assertion is just to verify, we are not setting this header (outside of mithril) for delete requests.
